IOPool/Input/src/RootDelayedReader.cc

```diff
@@ -24,8 +24,6 @@ namespace edm {
       : tree_(tree),
         filePtr_(filePtr),
         nextReader_(),
-        resourceAcquirer_(inputType == InputType::Primary ? new SharedResourcesAcquirer()
-                                                          : static_cast<SharedResourcesAcquirer*>(nullptr)),
```

Comment:
It is not related to ProcessBlock (I could have made a separate PR or ignored this). As I reviewed the code to understand what needed to be changed for ProcessBlock I noticed this. It is a useless line of code that serves no purpose. I just did a little cleanup and deleted it. I remember staring at it for a while trying to understand why it was there.

The RootDelayedReader constructor is below. Two cases:

1. Not primary. resourceAcquirer_ is a unique_ptr so it will be set to nullptr anyway. Don't need this line.

2. Primary. 5 lines later resourceAcquirer_ is reset to point at another object created with make_unique. The SharedResourcesAcquirer constructor and destructor do not have any side effects. The deleted line is just a wasted allocation and deallocation that serves no purpose.

```
  RootDelayedReader::RootDelayedReader(RootTree const& tree, std::shared_ptr<InputFile> filePtr, InputType inputType)
      : tree_(tree),
        filePtr_(filePtr),
        nextReader_(),
          RootDelayedReader::RootDelayedReader(RootTree const& tree, std::shared_ptr<InputFile> filePtr, InputType inputType)
      : tree_(tree),
        filePtr_(filePtr),
        nextReader_(),
        resourceAcquirer_(inputType == InputType::Primary ? new SharedResourcesAcquirer()
                                                          : static_cast<SharedResourcesAcquirer*>(nullptr)),
        inputType_(inputType),
        wrapperBaseTClass_(TClass::GetClass("edm::WrapperBase")) {
    if (inputType == InputType::Primary) {
      auto resources = SharedResourcesRegistry::instance()->createAcquirerForSourceDelayedReader();
      resourceAcquirer_ = std::make_unique<SharedResourcesAcquirer>(std::move(resources.first));
      mutex_ = resources.second;
    }
  }
(inputType == InputType::Primary ? new SharedResourcesAcquirer()
                                                          : static_cast<SharedResourcesAcquirer*>(nullptr)),
        inputType_(inputType),
        wrapperBaseTClass_(TClass::GetClass("edm::WrapperBase")) {
    if (inputType == InputType::Primary) {
      auto resources = SharedResourcesRegistry::instance()->createAcquirerForSourceDelayedReader();
      resourceAcquirer_ = std::make_unique<SharedResourcesAcquirer>(std::move(resources.first));
      mutex_ = resources.second;
    }
  }
```

Reply:
Thanks. A separate PR (or commit) would have been clearer, but it is not that big of a deal to include here.

Reply:
Move this deletion to a separate PR. #34406